### ISSUE TITLE
Ignore selection based on pattern

### ISSUE THREAD

**Issue by fdw** [mrichar1/clipster#64]
I'm using clipster with rofi and loving it so far. So, thank you very much for your work :)

I recently noticed that I have a lot of entries that follow a certain pattern and that I don't want to save. However, I cannot find a configuration value to ignore a selection change based on a pattern. Have I overlooked something? Would it be possible to implement something like this?

Thanks again!

**Comment by mrichar1** [comment]
Hi - sorry for the delay in looking at this!

I've added an `ignore_patterns` option - if this is enabled, then regular expression patterns will be read from `data_dir/ignore_patterns` (usually `.local.share/clipster/ignore_patterns`) one-per-line.

So for example:

```
politic
^cat(s)?$
```

Will ignore all selections containing the string `politic` and `cat` or `cats` (but not `catches`).

Let me know if this is enough to solve your issue, or if you encounter any problems with this change. Thanks!

**Comment by fdw** [comment]
Hi @mrichar1,
I haven't tested it yet, but it looks perfect! Thank you very much! 😊

Do you think it would be a good idea to make the path/filename of the ignore-file configurable? You could even fold that into the `ignore_patterns` option - if `ignore_patterns_from` is set, load the patterns from the option's parameter. If it is not set, no patterns are ignored.

**Comment by mrichar1** [comment]
clipster follows the Freedesktop XDG specifications: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

Thus the `history`, `patterns` (for `extract_patterns`) and now `ignore_patterns` files are all stored in either `$XDG_DATA_HOME/clipster` (or `~/.local/share/clipster` if this is unset).

So you can override the data directory by setting the `$XDG_DATA_HOME` environment variable when launching the clipster daemon to provide a different directory for these files.

I'd probably only consider changing how this works if there was a good reason to split the locations for the pattern and history files.

**Comment by fdw** [comment]
This was exactly my reason for asking: In my opinion, these patterns should be in `$XDG_CONFIG_HOME` (seeing as they're config files) rather than `$XDG_DATA_HOME`. However, I can live with both 😊

**Comment by mrichar1** [comment]
You're right actually - I'm not sure why they're in `$XDG_DATA_HOME` when they're config, unlike `history` `socket` `pid` etc which are created by the daemon.

I'll open a new issue for this. Thanks!

**Comment by mrichar1** [comment]
Following on from #65 - in refactoring `conf_dir` I decided after all to add `*_file` config options for `extract_patterns` and `ignore_patterns` (with sensible defaults) to match history, socket etc. I decided to keep the enabling and file paths options separate, to make setting defaults and toggling state easier without having to remove files etc. The default paths are now under `conf_dir` which in turn inherits from`$XDG_CONFIG_HOME`, as you correctly pointed out.

Let me know whether this refactored version solves your original issue, and lets you override the config as requested (comments here or in #65 as appropriate). Thanks!
